tests/structured_conf/data/attr_classes.py

```diff
@@ -22,8 +22,18 @@ class StructuredWithInvalidField:
 
 @attr.s(auto_attribs=True)
 class User:
-    name: str
-    age: int
```

Comment:
This did not result in an error before when constructing a Structured Config from User?

Reply:
No, it gave an error when creating new tests with it.

Reply:
so I am not sure, was it erroring out or not?
if it's "sometimes", when does it and when doesn't it?

Reply:
so when I tested the code it gave me an error:
```
test_merge.py:160: in <module>
    [Users, {"name2user": {"joe": User(name="joe")}}],
E   TypeError: __init__() missing 1 required positional argument: 'age'
```

Reply:
and this one:
```
test_merge.py:180: in <module>
    [{"user": DictConfig(ref_type=User, content=User)}, {"user": Group}],
../omegaconf/dictconfig.py:76: in __init__
    self._set_value(content)
../omegaconf/dictconfig.py:551: in _set_value
    data = get_structured_config_data(value)
../omegaconf/_utils.py:301: in get_structured_config_data
    return get_dataclass_data(obj)
../omegaconf/_utils.py:234: in get_dataclass_data
    _raise_missing_error(obj, name)
../omegaconf/_utils.py:177: in _raise_missing_error
    f"Missing default value for {get_type_of(obj).__name__}.{name}, to indicate "
E   ValueError: Missing default value for User.name, to indicate default must be populated later use OmegaConf.MISSING
```

Reply:
this was after adding the new changes

Reply:
ok. a bit weird that this test was there and it didn't cause issues until now.

Reply:
Okay, so it seems it's happening when using User in another dataclass (UserList, UserDict). When retrieving dataclass data. 
![image](https://user-images.githubusercontent.com/30913090/90415052-daec3300-e0b0-11ea-9c9e-1bce200a11c0.png)

Reply:
the class User were not as a Structured config by the tests before?
(which is when the dataclass data is being retrieved).